File: android/src/main/java/org/maplibre/reactnative/components/styles/sources/MLRNRasterSourceManager.java
package org.maplibre.reactnative.components.styles.sources;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.annotations.ReactProp;

import java.util.Map;

import javax.annotation.Nonnull;

public class MLRNRasterSourceManager extends MLRNTileSourceManager<MLRNRasterSource> {
    public static final String REACT_CLASS = "MLRNRasterSource";

    public MLRNRasterSourceManager(ReactApplicationContext reactApplicationContext) {
        super(reactApplicationContext);
    }

    @Nonnull
    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Nonnull
    @Override
    protected MLRNRasterSource createViewInstance(@Nonnull ThemedReactContext reactContext) {
        return new MLRNRasterSource(reactContext);
    }

    @ReactProp(name="tileSize")
    public void setTileSize(MLRNRasterSource source, int tileSize) {
        source.setTileSize(tileSize);
    }

    @Override
    public Map<String, String> customEvents() {
        return null;
    }
}

File: android/src/main/java/org/maplibre/reactnative/components/sources/tilesources/rastersource/MLRNRasterSourceManager.kt
package org.maplibre.reactnative.components.sources.tilesources.rastersource

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp
import org.maplibre.reactnative.components.sources.tilesources.MLRNTileSourceManager

class MLRNRasterSourceManager(context: ReactApplicationContext) :
    MLRNTileSourceManager<MLRNRasterSource>(context) {

    companion object {
        const val REACT_CLASS: String = "MLRNRasterSource"
    }

    override fun getName(): String = REACT_CLASS

    override fun createViewInstance(themedReactContext: ThemedReactContext): MLRNRasterSource {
        return MLRNRasterSource(themedReactContext)
    }

    @ReactProp(name = "tileSize")
    fun setTileSize(source: MLRNRasterSource, tileSize: Int) {
        source.tileSize = tileSize
    }
}
